#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.3259570505372824 -0.23834391296544902 -0.9148465337734816 4.320743721091272
  position 1.2151695427183402 -3.4402580637775055 2.89895149752306
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Robot {
  translation -0.197018 -0.597019 0.0426906
  rotation 0.00027068398963044695 5.7858297783523555e-05 0.9999999616912968 0.0113241
  children [
    DEF Hexapod Shape {
      appearance PBRAppearance {
        baseColor 0.513726 0.976471 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.5 0.1 0.05
      }
    }
    DEF H_R0 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.017092715963728567
        axis 0 0 1
        anchor -0.2 -0.05 0
      }
      device [
        RotationalMotor {
          name "r0_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation -0.2 -0.05 0
        rotation 0 0 -1 0.01709271596372661
        children [
          DEF Leg_shape Shape {
            appearance DEF Leg_app PBRAppearance {
              baseColor 1 0.666667 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.06
              radius 0.02
            }
          }
          DEF Right_Extension Transform {
            translation 0 -0.06 0
            children [
              DEF Extension_shape Shape {
                appearance DEF Extension_app PBRAppearance {
                }
                geometry Box {
                  size 0.02 0.1 0.02
                }
              }
            ]
          }
          DEF Right_Foot Transform {
            translation 0 -0.11 0
            children [
              DEF Foot_shape Shape {
                appearance DEF Foot_App PBRAppearance {
                  baseColor 0 0 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.06
                  radius 0.02
                }
              }
            ]
          }
          DEF R0_sj SliderJoint {
            jointParameters JointParameters {
              position -0.017102219551551834
              axis 0 0 -1
              minStop -0.02
              maxStop 0.02
            }
            device [
              DEF R0 LinearMotor {
                name "r0_leg"
                minPosition -0.02
                maxPosition 0.02
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 -0.11 -0.012897780448448169
              rotation 1 0 0 1.57
              children [
                DEF Finger_trans Transform {
                  rotation 1 0 0 1.57
                  children [
                    DEF Finger_shape Shape {
                      appearance DEF Finger_app PBRAppearance {
                        baseColor 1 0 0
                        roughness 1
                        metalness 0
                      }
                      geometry Cylinder {
                        height 0.06
                        radius 0.01
                      }
                    }
                  ]
                }
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        boundingObject USE Right_Foot
        physics Physics {
        }
      }
    }
    DEF H_R1 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.01709227251458481
        axis 0 0 1
        anchor 0 -0.05 0
      }
      device [
        RotationalMotor {
          name "r1_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation 0 -0.05 0
        rotation 0 0 -1 0.017092272514584096
        children [
          USE Leg_shape
          USE Right_Extension
          USE Right_Foot
          DEF R1_sj SliderJoint {
            jointParameters JointParameters {
              position -0.017102221503154866
              axis 0 0 -1
              minStop -0.02
              maxStop 0.02
            }
            device [
              LinearMotor {
                name "r1_leg"
                minPosition -0.02
                maxPosition 0.02
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 -0.11 -0.01289777849684513
              rotation 1 0 0 1.57
              children [
                USE Finger_trans
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        name "solid(1)"
        boundingObject USE Right_Foot
        physics Physics {
        }
      }
    }
    DEF H_R2 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.01709224927181634
        axis 0 0 1
        anchor 0.2 -0.05 0
      }
      device [
        RotationalMotor {
          name "r2_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation 0.2 -0.05 0
        rotation 0 0 -1 0.017092249271813444
        children [
          USE Leg_shape
          USE Right_Extension
          USE Right_Foot
          DEF R2_sj SliderJoint {
            jointParameters JointParameters {
              position -0.017102223961600896
              axis 0 0 -1
              minStop -0.02
              maxStop 0.02
            }
            device [
              LinearMotor {
                name "r2_leg"
                minPosition -0.02
                maxPosition 0.02
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 -0.11 -0.012897776038399106
              rotation 1 0 0 1.57
              children [
                USE Finger_trans
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        name "solid(2)"
        boundingObject USE Right_Foot
        physics Physics {
        }
      }
    }
    DEF H_L0 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.017092460071344972
        axis 0 0 1
        anchor -0.2 0.05 0
      }
      device [
        RotationalMotor {
          name "l0_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation -0.2 0.05 0
        rotation 0 0 -1 0.017092460071348112
        children [
          USE Leg_shape
          DEF Left_Extension_trans Transform {
            translation 0 0.06 0
            children [
              USE Extension_shape
            ]
          }
          DEF Left_Foot_trans Transform {
            translation 0 0.11 0
            children [
              USE Foot_shape
            ]
          }
          DEF L0_sj SliderJoint {
            jointParameters JointParameters {
              position -0.017102172613967688
              axis 0 0 -1
              minStop -0.02
              maxStop 0.02
            }
            device [
              LinearMotor {
                name "l0_leg"
                minPosition -0.02
                maxPosition 0.02
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 0.11 -0.012897827386032321
              rotation 1 0 0 1.57
              children [
                USE Finger_trans
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        name "solid(3)"
        boundingObject USE Left_Foot_trans
        physics Physics {
        }
      }
    }
    DEF H_L1 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.01709228015406694
        axis 0 0 1
        anchor 0 0.05 0
      }
      device [
        RotationalMotor {
          name "l1_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation 0 0.05 0
        rotation 0 0 -1 0.01709228015407278
        children [
          USE Leg_shape
          USE Left_Extension_trans
          USE Left_Foot_trans
          DEF L1_sj SliderJoint {
            jointParameters JointParameters {
              position -0.017102176262918233
              axis 0 0 -1
              minStop -0.2
              maxStop 0.2
            }
            device [
              LinearMotor {
                name "l1_leg"
                minPosition -0.2
                maxPosition 0.2
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 0.11 -0.012897823737081766
              rotation 1 0 0 1.57
              children [
                USE Finger_trans
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        name "solid(4)"
        boundingObject USE Left_Foot_trans
        physics Physics {
        }
      }
    }
    DEF H_L2 HingeJoint {
      jointParameters HingeJointParameters {
        position -0.017092665822706896
        axis 0 0 1
        anchor 0.2 0.05 0
      }
      device [
        RotationalMotor {
          name "l2_motor"
          maxVelocity 5
          minPosition -0.7
          maxPosition 0.7
        }
      ]
      endPoint Solid {
        translation 0.2 0.05 0
        rotation 0 0 -1 0.017092665822711257
        children [
          USE Leg_shape
          USE Left_Extension_trans
          USE Left_Foot_trans
          DEF L2_sj SliderJoint {
            jointParameters JointParameters {
              position -0.01710218043423583
              axis 0 0 -1
              minStop -0.2
              maxStop 0.2
            }
            device [
              LinearMotor {
                name "l2_leg"
                minPosition -0.2
                maxPosition 0.2
                maxForce 25
              }
            ]
            endPoint Solid {
              translation 0 0.11 -0.012897819565764176
              rotation 1 0 0 1.57
              children [
                USE Finger_trans
              ]
              boundingObject USE Finger_trans
              physics Physics {
              }
            }
          }
        ]
        name "solid(5)"
        boundingObject USE Left_Foot_trans
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.25 -0.03 -0.01
      scale 0.1 0.1 0.1
      children [
        DEF dsensor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "ds1"
      boundingObject USE dsensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.3 1000 0
      ]
    }
    DistanceSensor {
      translation 0.25 0.03 -0.01
      scale 0.1 0.1 0.1
      children [
        DEF dsensor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "ds2"
      boundingObject USE dsensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.3 1000 0
      ]
    }
  ]
  boundingObject USE Hexapod
  physics Physics {
  }
  controller "my_controller"
}
RectangleArena {
  floorSize 3 3
}
